build.gradle

```diff
@@ -279,6 +279,8 @@ configure(project.fineractJavaProjects) {
     sourceSets.main.output.resourcesDir = sourceSets.main.java.outputDir
     sourceSets.test.output.resourcesDir = sourceSets.test.java.outputDir
 
+    check.dependsOn('cucumber')
```

Comment:
Not sure if I follow here... doesn't this mean that the cucumber tests kick in when we do linting with Spotless etc.? Linting and testing being two different build lifecycle stages?

Reply:
Good catch Aleks. The reason I moved the cucumber execution here is because originally if I wanted to execute simple unit tests in fineract-provider, it was first running ALL the cucumber tests. And considering they are Spring tests, it's quite slow. For a single unit test it was taking 30-60 seconds to run which is way too long.

Moving the cucumber execution to depend on the check task means that the unit tests can be easily executed without the cucumber tests being run beforehand.

Also, this change is in line with the official Gradle recommendation:

> check: Aggregate task that performs verification tasks, such as running the tests. Some plugins add their own verification tasks to check. You should also attach any custom Test tasks to this lifecycle task if you want them to execute for a full build. This task is added by the Base Plugin.

Source: https://docs.gradle.org/current/userguide/java_plugin.html

Reply:
alright... interesting.